src/SpinAssumePass.cpp

```diff
@@ -186,7 +186,12 @@ bool SpinAssumePass::assumify_loop(llvm::Loop *l,llvm::LPPassManager &LPM){
   }
   llvm::Value *cond = BI->getCondition();
   llvm::MDNode *MD = BI->getMetadata("dbg");
+#if LLVM_VERSION_MAJOR > 15
+  auto lastInstIt = --EB->end();
+  EB->erase(lastInstIt,EB->end());
```

Comment:
You could use `std::prev`

If  the new code works then why not use it on all llvm versions? The less ifdeffery, the better.